libclamav/filetypes.c

```diff
@@ -277,7 +277,8 @@ cli_file_t cli_determine_fmap_type(fmap_t *map, const struct cl_engine *engine,
     unsigned char buffer[MAGIC_BUFFER_SIZE];
     const unsigned char *buff;
     unsigned char *decoded;
-    int bread, sret;
+    int bread;
+    cli_file_t sret;
```

Comment:
The lines 303 -> 308 (below) should use a different variable than `sret` that IS an `int`, because `cli_memcpy` returns an int. May also be worth considering having it return a bool ... but would have to evaluate where else `cli_memcpy` is used.:
```c
        sret = cli_memcpy(buffer, buff, bread);
        if (sret) {
            cli_errmsg("cli_determine_fmap_type: fileread error!\n");
            return CL_TYPE_ERROR;
        }
        sret = 0;
```